components/learn/modules/ROOT/pages/on-dapps.adoc

```diff
@@ -1,8 +1,160 @@
+
+  * <<network, Connecting to the Network>>
+  * <<hot-loader, Hot-loading your Contracts>>
+  * <<starter-kit, Kickstarting a Dapp using Starter Kits>>
+
+If you are in a rush, just skip over to <<starter-kit, the last section>>!
+
+[[network]]
+== Connecting to the Network
+
+When building a dapp, the first step is to connect to the Ethereum network. This usually involves the following steps:
+
+  . If the user is running Metamask, use the injected web3 provider to access the network, and monitor for network or account changes
+  . If not, fall back to a connection to a public Infura node
+
+To avoid having to code this logic on every dapp, we built https://github.com/OpenZeppelin/openzeppelin-network.js[`@openzeppelin/network`] to take care of it. It ships with a `useWeb3` https://reactjs.org/docs/hooks-intro.html[React hook] for getting a `web3Context` based on the logic above. This `web3Context` object provides access to a connected `web3` instance, network information, the user accounts, a function to https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1102.md[request account access], and more.
+
+The following snippet shows a small React app component that connects to the network using Metamask (falling back to Infura if not found), prompts the user for account access, and shows the user balance.
+
+```js
+import React, { useCallback, useState, useEffect } from 'react';
+import { useWeb3 } from '@openzeppelin/network/react';
+
+function App() {
+  // Get web3 context via @openzeppelin/network/react hook
+  const web3Context = useWeb3(`wss://mainnet.infura.io/ws/v3/${PROJECT_ID}`);
+  const { lib: web3, networkId, accounts, providerName } = web3Context;
+
+  // Methods for requesting accounts access
+  const requestAuth = (web3Context) => web3Context.requestAuth();
+  const requestAccess = useCallback(() => requestAuth(web3Context), []);
+
+  // Querying account balance
+  const [balance, setBalance] = useState(0);
+  const getBalance = useCallback(async () => {
+    setBalance(
+      accounts && accounts.length > 0 
+      ? (await web3.eth.getBalance(accounts[0]))
+      : 'Unknown')
+  }, [accounts, web3.eth, web3.utils]);
+  useEffect(() => {
+    getBalance();
+  }, [accounts, getBalance, networkId]);
+
+  // Show all information to the user
+  return (
+    <div>
+      <div>Network: {networkId || 'No connection'}</div>
+      <div>Your address: {accounts ? accounts[0] : 'Unknown'}</div>
+      
+      {accounts && accounts.length ? (
+        <div>Accounts access granted</div>
+      ) : !!networkId && providerName !== 'infura' ? (
+        <button onClick={requestAccess}>Request Access</button>
+      ) : (
+        <div>No accounts access</div>
+      )}
+    </div>
+  );
+}
+```
+
```

Comment:
I feel like this is lacking some sort of conclusion, perhaps something to the effect of 'we just made an app that connects to the ethereum network and integrates with metamask without having to worry about these details'.